superset/db_engine_specs/mssql.py

```diff
@@ -76,3 +77,16 @@ def get_sqla_column_type(cls, type_: str) -> Optional[TypeEngine]:
             if regex.match(type_):
                 return sqla_type
         return None
+
+    @classmethod
+    def modify_url_for_impersonation(
+        cls, url: URL, impersonate_user: bool, username: Optional[str]
+    ) -> None:
+        """
+        Modify the SQL Alchemy URL object with the user to impersonate if applicable.
+        :param url: SQLAlchemy URL object
+        :param impersonate_user: Flag indicating if impersonation is enabled
+        :param username: Effective username
+        """
+        if impersonate_user and username is not None:
+            url.query["conn_properties"] = "EXECUTE AS USER = '{}'".format(username)
```

Comment:
We tend to use f-strings, e.g. `abc = f"variable goes {here}"`.

Reply:
You are right, `pymssql` is deprecated and this feature is limited to that adapter. I've analyzed first `pyodbc` but I didn't find a way to send SQL queries to the server upon the connection establishment. I added logic to check the driver being used and applied this logic only for `pymssql`. For other adapters I just don't know if we should raise an exception or just add the default behaviour, which is replacing the connection string username by the logged in user's. What do you think?

Reply:
Hi @villebro, did you have a chance to think on my question? Thanks!